filament/backend/src/vulkan/VulkanContext.cpp

```diff
@@ -357,7 +359,17 @@ void createSwapChain(VulkanContext& context, VulkanSurfaceContext& surfaceContex
         .imageExtent = size,
         .imageArrayLayers = 1,
         .imageUsage = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT,
+
+        // TODO: Setting the preTransform to IDENTITY means we are letting the Android Compositor
```

Comment:
With Vulkan we definitely need to handle pre-rotation correctly otherwise we'll get horrible performance in one of the display orientations.

Reply:
@romainguy According to the article, we would need to adjust our MVP as well as all our dFdx/dFdy calls in GLSL, which seems a bit painful.  :-\  How is this done in OpenGL?

Reply:
It's done magically by the driver on Android. The Graphics team can help us to do this right, and maybe glslang/spirv-* have support for this? Changing derivatives would be a major pain. Wouldn't that mean calling the *Grad() variants of texture calls too?

Reply:
I don't think we need to do anything because everything happens on the GPU and textures are all swizzled, I don't see why 90 rotation of a swizzled texture would be slower. On the Android UI things are different because the gralloc buffers are not swizzled (because they need to be read by the HWC).

The only case where this would be a problem is when we draw directly into the swapchain -- but that basically never happens.

Reply:
Right but our final buffer is not swizzled, the one coming from the Surface? Shouldn't we handle pre-rotation there at least?

Reply:
Yes, yes. But I meant this will be much easier, because it's a post-process pass, often just a blit. So derivatives shouldn't be a problem, nor the MVP.